tests/test_expression.py

```diff
@@ -1306,3 +1318,10 @@ def test_expr_str() -> None:
                        3))"""
         ),
     )
+
+
+def test_call_str() -> None:
+    assert str(Add(Number(1), Call("Test", []))) == "1 + Test"
+    assert str(Add(Number(1), -Call("Test", []))) == "1 - Test"
```

Comment:
A unit test for `Call` should only depend on the implementation of `Call`. This unit test will fail if the implementation of `Add` changes. (Many tests depend on `Number` and `Variable`. I think that's a compromise. `Add` does not have an as simple implementation as these two.)

Reply:
The unit test does specifically test the behaviour of `Call` inside a `BinExpr` so it has to depend on both.

Reply:
`Add` is not a `BinExpr`, but an `AssExpr`. I see that `Add.__str__` handles instances of `Name` differently. But still: If you want to test the handling of `Name` in `Add.__str__`, you should do that in `test_add_str`.

Reply:
Fixed in https://github.com/Componolit/RecordFlux/pull/390/commits/92983bdc4d9c64b48d67b21c245d402eb83d0ba9.